pkg/specs/githubclient/client.go

```diff
@@ -165,6 +167,21 @@ func (g *GithubClient) downloadAndExtractFiles(
 	return nil
 }
 
+func includeDirOrFile(currentPath, includePath string, currentIsFile bool) bool {
+	splitInclude := strings.Split(includePath, "/")
+	includeIsFile := strings.Contains(splitInclude[len(splitInclude)-1], ".")
+
+	if includeIsFile {
+		if currentIsFile {
+			return currentPath == includePath
+		}
+
+		return strings.HasPrefix(includePath, currentPath)
+	}
+
+	return strings.HasPrefix(currentPath, includePath)
```

Comment:
I feel like this will break on a couple of situations, the most obvious one being when you have a directory name that is a substring of another directory name:
currentPath `dir/file2/blah`, includePath `dir/file`

We might have written code for this elsewhere in Ship

Reply:
We could also just not try and special case the "singleton file" bit here and instead always create the parent directory for files before creating the files
That would require the "mkdir -p" equivalent to exist, of course